packages/flutter/lib/src/material/slider.dart

```diff
@@ -540,13 +551,13 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
   }
 
   void _handleDragStart(double value) {
-    assert(widget.onChangeStart != null);
-    widget.onChangeStart!(_lerp(value));
```

Comment:
Do we know the history of this assert? Is it here just to protect the reference in the call below? If so, yeah it would be good to remove it as you have. But was it actually a requirement of the widget to not have this null (which seems odd)?  If so it should be moved to the widget constructor, no? Same for the similar assert in `_handleDragEnd`.

Reply:
It was just to defend against an error being introduced in the original code on lines 751,752. No longer needed.